pkg/scm/github.go

```diff
@@ -56,22 +57,72 @@ func (g *Github) PublishIssuesReviewFor(ctx context.Context, issues []sonarqube.
 		reviewEvent = REVIEW_EVENT_COMMENT
 	}
 
+	// Convert PR number into int
+	prNumber, err := strconv.Atoi(pr.Key)
+	if err != nil {
+		return errors.Wrap(err, "failed to convert PR number to int")
+	}
+
+	// Parse PR path
+	ghPath, err := parseGithubPath(pr.URL)
+	if err != nil {
+		return errors.Wrap(err, "failed to parse github path")
+	}
+
+	// Fetch PR diffs
+	ghDiff, _, err := g.client.PullRequests.GetRaw(ctx, ghPath.Owner, ghPath.Repo, prNumber, github.RawOptions{Type: github.Diff})
+	if err != nil {
+		return errors.Wrap(err, "failed to get raw PR")
+	}
+
+	// Parse diffs
+	fileDiffs, err := diff.ParseMultiFileDiff([]byte(ghDiff))
+	if err != nil {
+		return errors.Wrap(err, "failed to parse diff")
+	}
+
+	diffMap := make(map[string][]*diff.Hunk)
+	for i := range fileDiffs {
+		fileName := fileDiffs[i].OrigName[2:]
+		diffMap[fileName] = fileDiffs[i].Hunks
+	}
+
 	comments := make([]*github.DraftReviewComment, 0)
 
 	// Create a comment for each issue
 	for _, issue := range issues {
 		side := "RIGHT"
 		message := issue.MarkdownMessage(g.sonar.Root)
 		filePath := issue.FilePath()
-		line := issue.Line
+		lineNumber := issue.Line
+
+		// Skip if current issue is not part of the PR diff
+		hunks, ok := diffMap[filePath]
+		if !ok {
+			continue
+		}
 
-		comment := &github.DraftReviewComment{
-			Path: &filePath,
-			Body: &message,
-			Side: &side,
-			Line: &line,
+		for _, hunk := range hunks {
+			if lineNumber < int(hunk.OrigStartLine) || lineNumber < int(hunk.NewStartLine) {
+				continue
+			}
+			if lineNumber > int(hunk.OrigStartLine+hunk.OrigLines) || lineNumber > int(hunk.NewStartLine+hunk.NewLines) {
+				continue
+			}
+
+			comment := &github.DraftReviewComment{
+				Path: &filePath,
+				Body: &message,
+				Side: &side,
+				Line: &lineNumber,
+			}
+			comments = append(comments, comment)
 		}
-		comments = append(comments, comment)
+
+	}
+
+	if len(comments) == 0 {
+		return errors.New("failed to find relevant issues")
 	}
 
 	body := fmt.Sprintf(`:wave: Hey, I added %d comments about your changes, please take a look :slightly_smiling_face:`, len(issues))
```

Comment:
@herlon214 I forgot to update the `I added %d comments about your changes` to use `len(comments)` as now the `len(issues)` >= `len(comments)`

Reply:
Good catch

Reply:
Fixed on v0.5.1, thanks @adenisonafifi